scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/signal/SignalApiImpl.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.signal;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.awaitility.Awaitility;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+import org.ow2.proactive.scheduler.synchronization.InvalidChannelException;
+import org.ow2.proactive.scheduler.synchronization.SynchronizationInternal;
+import org.ow2.proactive.scheduler.synchronization.SynchronizationWrapper;
+
+
+/**
+ * SignalApiImpl acts as a high level signal service that uses the synchronization API.
+ * It enables workflow tasks handle signals in an easy manner.
+ *
+ * It implements the methods defined in the interface @see org.ow2.proactive.scheduler.signal.SignalApi.
+ *
+ * @author ActiveEon Team
+ * @since 24/11/2020
+ */
+public class SignalApiImpl implements SignalApi {
+
+    private static final Logger logger = Logger.getLogger(SignalApiImpl.class);
+
+    private static final String LOG_WARNING_CONSTANT = " of the job ";
```

Comment:
add `of the job` in the logger directly. It is not that useful to have a constant here.

Reply:
I already added the text in the logger directly, but Sonar told me to add a constant instead of repeating the same statement many times :)